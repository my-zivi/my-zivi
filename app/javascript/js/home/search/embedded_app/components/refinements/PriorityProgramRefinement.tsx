import { ToggleRefinement } from 'react-instantsearch-dom';
import React from 'preact/compat';
import defaultTranslations from './defaultTranslations';

const PriorityProgramRefinement: React.FunctionComponent = () => (
  <ToggleRefinement
    attribute="priority_program"
    label={MyZivi.translations.search.refinements.priorityProgramLabel}
    limit={5}
    value={true}
    translations={defaultTranslations}
  />
);

export default PriorityProgramRefinement;
