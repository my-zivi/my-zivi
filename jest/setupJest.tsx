/* eslint-disable @typescript-eslint/no-explicit-any */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-preact-pure';
import React, { h, Fragment } from 'preact';
import '../app/javascript/js/shared/globals.d';

(global as any).h = h;
(global as any).Fragment = Fragment;

const globalConfig: MyZiviGlobal = {
  locale: 'de',
  translations: {
    servicesOverview: {
      title: 'My Services Overview',
      errors: {
        cannotLoad: 'Konnte die EinsÃ¤tze nicht laden',
      },
    },
    search: {
      loadMore: 'string',
      noResultsFound: 'string',
      poweredBy: 'string',
      refinements: {
        canton: 'string',
        category: 'string',
        collapseButton: 'string',
        controls: {
          noResults: 'string',
          placeholder: 'string',
          reset: 'string',
          resetTitle: 'string',
          showLess: 'string',
          showMore: 'string',
          submitTitle: 'string',
        },
        language: 'string',
        languages: {
          french: 'string',
          german: 'string',
          italian: 'string',
        },
        month: {
          one: 'string',
          other: 'string',
        },
        minimumServiceMonths: 'string',
        priorityProgram: 'string',
        priorityProgramLabel: 'string',
      },
      ribbons: {
        featured: 'Feat.',
        featuredFull: 'Featured',
        new: 'Neu',
        urgent: 'Wichtig',
      },
      search: 'string',
      searchPlaceholder: 'string',
      statistics: 'string',
      searchView: {
        map: 'Karte',
        tiles: 'Raster',
      },
    } as any,
    loading: 'Laden...',
    error: 'Fehler',
  },
  paths: {
    servicesOverview: '/my/mock/services.json',
  },
  algolia: {
    apiKey: 'my-api-key',
    applicationId: 'my-application-id',
  },
};
(global as any).MyZivi = globalConfig;

jest.mock('@rails/ujs', () => jest.requireActual('js/shared/__mocks__/Rails'));

jest.mock('@fortawesome/react-fontawesome', () => {
  const component: React.FunctionComponent<any> = () => <></>;
  component.displayName = 'FontAwesomeIcon';
  return {
    FontAwesomeIcon: component,
  };
});

configure({ adapter: new Adapter() });
