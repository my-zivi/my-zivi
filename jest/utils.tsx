/* eslint-disable @typescript-eslint/no-explicit-any */
import { mount } from 'enzyme';
import { IApi } from 'js/shared/Api';
import preact from 'preact';
import { ApiContext } from 'js/shared/ApiProvider';
import React from 'preact/compat';
import { InstantSearch } from 'react-instantsearch-dom';
import { defer } from 'lodash';
import { mockClient } from '../app/javascript/js/tests/algolia/mocks';

type MountComponentType = (componentInstance: preact.JSX.Element, targetComponentName?: string) => any;
export const mountConsumer: MountComponentType = (componentInstance, targetComponentName) => {
  const root = mount(componentInstance);

  if (!targetComponentName) {
    return root;
  }

  return root.find(targetComponentName);
};

type FakeWrapInApiProviderType = (api: IApi) => (wrapped: preact.JSX.Element, component: string) => any;
export const fakeWrapInApiProvider: FakeWrapInApiProviderType = (api: IApi) => (
  (wrapped, component) => mountConsumer(
    (
      <ApiContext.Provider value={api}>
        {wrapped}
      </ApiContext.Provider>
    ),
    component,
  )
);

export const mountInstantSearchClient: MountComponentType = (componentInstance, targetComponentName?) => mountConsumer(
  <InstantSearch searchClient={mockClient} indexName="JobPosting">
    {componentInstance}
  </InstantSearch>,
  targetComponentName,
);

type DeferredCheck = (rendered: { update: () => void }, callback: () => void | Promise<void>) => Promise<void>;
export const deferredCheck: DeferredCheck = (rendered, callback) => new Promise<void>((resolve, reject) => {
  defer(() => {
    rendered.update();
    const result = callback();
    if (!result) {
      return resolve();
    }

    return result.then(resolve).catch(reject);
  });
});
