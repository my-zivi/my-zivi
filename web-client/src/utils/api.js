const BASE_URL = 'http://localhost:8000/api/';

// Is user logged in?
const isLoggedIn = () => {
  if (localStorage.getItem('jwtToken') !== null) {
    return true;
  } else {
    return false;
  }
};

// Is user an administrator?
export const isAdmin = () => {
  if (isLoggedIn()) {
    const jwtDecode = require('jwt-decode');
    const decodedToken = jwtDecode(localStorage.getItem('jwtToken'));
    return decodedToken.isAdmin;
  }
  return false;
};

// Verify that the fetched response is JSON
export function _verifyResponse(res) {
  let contentType = res.headers.get('content-type');

  if (contentType && contentType.indexOf('application/json') !== -1) {
    return res.json();
  } else {
    _handleError({ message: 'Response was not JSON' });
  }
}

const ApiService = { BASE_URL, isLoggedIn, isAdmin };
export default ApiService;
