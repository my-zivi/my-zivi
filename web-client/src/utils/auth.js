import moment from 'moment';
import jwtDecode from 'jwt-decode';
import Raven from 'raven-js';

// Is user logged in?
const isLoggedIn = () => {
  if (localStorage.getItem('jwtToken') !== null) {
    const decodedToken = jwtDecode(localStorage.getItem('jwtToken'));
    return moment.unix(decodedToken.exp).isAfter(moment());
  } else {
    return false;
  }
};

// Is user an administrator?
const isAdmin = () => {
  if (isLoggedIn()) {
    const decodedToken = jwtDecode(localStorage.getItem('jwtToken'));
    return decodedToken.isAdmin;
  }
  return false;
};

// Get logged in user id
const getUserId = () => {
  if (isLoggedIn()) {
    const decodedToken = jwtDecode(localStorage.getItem('jwtToken'));
    return decodedToken.sub;
  }
};

const setToken = token => {
  localStorage.setItem('jwtToken', token);
  Raven.setUserContext({ id: getUserId() });
};

// Verify that the fetched response is JSON
function _verifyResponse(res) {
  let contentType = res.headers.get('content-type');

  if (contentType && contentType.indexOf('application/json') !== -1) {
    return res.json();
  } else {
    _handleError({ message: 'Response was not JSON' });
  }
}

const Auth = { isLoggedIn, isAdmin, getUserId, setToken };
export default Auth;
