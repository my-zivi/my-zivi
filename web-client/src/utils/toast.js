import iziToast from 'izitoast/dist/js/iziToast.js';

function showSuccess(title, msg) {
  iziToast.show({
    id: 'save-success',
    theme: 'light',
    color: 'green',
    title: title,
    message: msg,
    position: 'topCenter',
    transitionIn: 'flipInX',
    transitionOut: 'flipOutX',
    progressBarColor: 'rgb(0, 255, 0)',
    layout: 2,
    timeout: 2000,
    onClose: function() {
      // console.info('onClose');
    },
  });
}

function showError(title, msg) {
  iziToast.show({
    id: 'save-success',
    theme: 'dark',
    color: 'red',
    title: title,
    message: msg,
    position: 'topCenter',
    transitionIn: 'flipInX',
    transitionOut: 'flipOutX',
    progressBarColor: 'rgb(255, 0, 0)',
    layout: 2,
    timeout: 4000,
    onClose: function() {
      // console.info('onClose');
    },
  });
}

// Export
const Toast = { showSuccess, showError };
export default Toast;
