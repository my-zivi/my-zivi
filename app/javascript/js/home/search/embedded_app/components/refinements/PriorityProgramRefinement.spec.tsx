import React from 'preact/compat';
import { deferredCheck, mountInstantSearchClient } from '../../../../../../../../jest/utils';
import PriorityProgramRefinement from './PriorityProgramRefinement';

describe('PriorityProgramRefinement', () => {
  const rendered = mountInstantSearchClient(<PriorityProgramRefinement />, 'PriorityProgramRefinement');

  it('renders correctly', () => deferredCheck(rendered, () => {
    expect(rendered).toMatchSnapshot();
  }));
});
