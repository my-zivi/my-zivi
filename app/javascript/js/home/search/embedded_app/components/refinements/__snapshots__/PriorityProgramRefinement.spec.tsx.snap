// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriorityProgramRefinement renders correctly 1`] = `
<PriorityProgramRefinement>
  <ConnectorWrapper
    attribute="priority_program"
    label="string"
    limit={5}
    translations={
      Object {
        "noResults": "string",
        "placeholder": "string",
        "resetTitle": "string",
        "showMore": [Function],
        "submitTitle": "string",
      }
    }
    value={true}
  >
    <Consumer>
      <Consumer>
        <AlgoliaToggle(ToggleRefinement)
          attribute="priority_program"
          contextValue={
            Object {
              "createHrefForState": [Function],
              "mainTargetedIndex": "JobPosting",
              "onInternalStateUpdate": [Function],
              "onSearchForFacetValues": [Function],
              "onSearchParameters": [Function],
              "onSearchStateChange": [Function],
              "store": Object {
                "getState": [Function],
                "setState": [Function],
                "subscribe": [Function],
              },
              "widgetsManager": Object {
                "getWidgets": [Function],
                "registerWidget": [Function],
                "update": [Function],
              },
            }
          }
          label="string"
          limit={5}
          translations={
            Object {
              "noResults": "string",
              "placeholder": "string",
              "resetTitle": "string",
              "showMore": [Function],
              "submitTitle": "string",
            }
          }
          value={true}
        >
          <ToggleRefinement
            attribute="priority_program"
            canRefine={false}
            className=""
            count={
              Object {
                "checked": null,
                "unchecked": null,
              }
            }
            createURL={[Function]}
            currentRefinement={false}
            label="string"
            limit={5}
            refine={[Function]}
            translations={
              Object {
                "noResults": "string",
                "placeholder": "string",
                "resetTitle": "string",
                "showMore": [Function],
                "submitTitle": "string",
              }
            }
            value={true}
          >
            <div
              className="ais-ToggleRefinement ais-ToggleRefinement--noRefinement"
            >
              <label
                className="ais-ToggleRefinement-label"
              >
                <input
                  checked={false}
                  className="ais-ToggleRefinement-checkbox"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="ais-ToggleRefinement-labelText"
                >
                  string
                </span>
              </label>
            </div>
          </ToggleRefinement>
        </AlgoliaToggle(ToggleRefinement)>
      </Consumer>
    </Consumer>
  </ConnectorWrapper>
</PriorityProgramRefinement>
`;
